data/src/test/java/org/apache/iceberg/data/DeleteReadTests.java

```diff
@@ -108,11 +112,8 @@ public void writeTestDataFile() throws IOException {
     records.add(record.copy("id", 122, "data", "g"));
 
     this.dataFile =
-        FileHelpers.writeDataFile(
-            table,
-            Files.localOutput(File.createTempFile("junit", null, temp.toFile())),
-            Row.of(0),
-            records);
+        FileHelpers.writeDataFile(table, Files.localOutput(temp), Row.of(0), records);
```

Comment:
please revert all changes to `DeleteReadTest`. Tests need a new temp file and your changes aren't providing that